test/celeritas/optical/Rayleigh.test.cc

```diff
@@ -0,0 +1,140 @@
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/OpticalRayleigh.test.cc
+//---------------------------------------------------------------------------//
+#include "celeritas/optical/model/RayleighInteractor.hh"
+#include "celeritas/phys/InteractorHostTestBase.hh"
+
+#include "celeritas_test.hh"
+
+namespace celeritas
+{
+namespace optical
+{
+namespace test
+{
+using namespace ::celeritas::test;
+
+//---------------------------------------------------------------------------//
+
+class RayleighInteractorTest : public InteractorHostTestBase
+{
+  protected:
+    void SetUp() override
+    {
+        inc_direction_ = {0, 0, 1};
+
+        // Check incident quantities are valid
+        sanity_check(inc_direction_, track_view().polarization());
+    }
+
+    void sanity_check(Real3 const& direction, Real3 const& polarization) const
+    {
+        // Check that direction and polarization are unit vectors
+        EXPECT_SOFT_EQ(1, norm(direction));
+        EXPECT_SOFT_EQ(1, norm(polarization));
+        // Check that direction and polarization are perpendicular
+        EXPECT_SOFT_EQ(0, dot_product(direction, polarization));
+    }
+
+    void sanity_check(Interaction const& interaction) const
+    {
+        // Interactions should always be scattering
+        EXPECT_EQ(Interaction::Action::scattered, interaction.action);
+        sanity_check(interaction.direction, interaction.polarization);
+    }
+
+    TrackView const& track_view() const
+    {
+        // Energy doesn't matter; initial polarization should be perpendicular
+        static TrackView view{units::MevEnergy{13e-6}, {1, 0, 0}};
+        return view;
+    }
+
+    Real3 inc_direction_;
+};
+
+TEST_F(RayleighInteractorTest, basic)
+{
+    int const num_samples = 4;
+
+    RayleighInteractor interact{track_view(), inc_direction_};
+
+    auto& rng_engine = this->rng();
+
+    std::vector<real_type> dir_angle;
+    std::vector<real_type> pol_angle;
+
+    for ([[maybe_unused]] int i : range(num_samples))
+    {
+        Interaction result = interact(rng_engine);
+        this->sanity_check(result);
+
+        dir_angle.push_back(dot_product(result.direction, inc_direction_));
+        pol_angle.push_back(
+            dot_product(result.polarization, track_view().polarization()));
+    }
+
+    static double const expected_dir_angle[] = {-0.38366589898599,
+                                                -0.18253767807648,
+                                                0.42140775018143,
+                                                -0.15366976713254};
+    static double const expected_pol_angle[] = {
+        0.46914077892327, 0.31484033374691, 0.82815845476385, 0.50516788333603};
+
+    EXPECT_VEC_SOFT_EQ(expected_dir_angle, dir_angle);
+    EXPECT_VEC_SOFT_EQ(expected_pol_angle, pol_angle);
+}
+
+TEST_F(RayleighInteractorTest, stress_test)
+{
+    int const num_samples = 1'000;
```

Comment:
Fancy! First use of the digits separator in our code 😄 we should consider doing this for the other large integers in use...